src/group_theory/subgroup.lean

```diff
@@ -334,6 +334,98 @@ theorem multiplicative.normal_subgroup_iff [add_group α]
     @normal_add_subgroup.mk α _ _ (multiplicative.is_subgroup_iff.1 h₁) @h₂,
   λ h, by resetI; apply_instance⟩
 
+
+/-The normal closure of a set s is the subgroup closure of all the conjugates of
+elements of s. It is the smallest normal subgroup containing s. -/
+
+namespace group
+variables {s : set α} [group α]
+
+/-- Given an element a, conjugates a is the set of conjugates. -/
+def conjugates (a : α) : set α := {b | is_conj a b}
+
+lemma mem_conjugates_self {a : α} : a ∈ conjugates a := is_conj_refl _
+
+/-- Given a set s, conjugates_of_set s is the set of all conjugates of
+the elements of s. -/
+def conjugates_of_set (s : set α) : set α := ⋃ a ∈ s, conjugates a
+
+lemma mem_conjugates_of_set_iff {x : α} : x ∈ conjugates_of_set s ↔ ∃ a : α, a ∈ s ∧ is_conj a x :=
+  set.mem_bUnion_iff
```

Comment:
Style remark: If the type starts on a new line, it is indented. Proofs always start at the left margin. Please fix this (multiple places).

Reply:
Done.